### PR TITLE
Simplify WorkFlow reading, fix workflow validation and unit testing

### PR DESCRIPTION
**Pull request by mhoffrog**
## Changes
s. [CHANGELOG.md](https://github.com/mavenplugins/maven-cdi-plugin-utils/blob/master/CHANGELOG.md)

## Summary by CodeRabbit

* Documentation
  * Added 4.0.2 release notes and updated changelog links.
* Deprecations
  * Introduced deprecated stream-based workflow APIs and new trimmed-line workflow utilities.
* Chores
  * Bumped Guava to 33.5.0-jre and updated build metadata.
* Security
  * Addressed a GitHub Actions security warning.
* Tests
  * Updated tests to use new utilities, added a test for missing workflows, and adjusted signatures.
* Bug Fixes
  * Improved workflow parsing/validation, parallel-block handling, and try/finally edge-case behavior.

### CODE REVIEW

File: src/main/java/com/itemis/maven/plugins/cdi/internal/util/workflow/WorkflowValidator.java [hunk 1]
```diff
@@ -15,33 +10,35 @@
...
+              "There is a finally-block opening without a related try-block. Processed line was: '" + line + "'");
         }
       }
-    } catch (IOException e) {
-      throw new RuntimeException("Unable to read the workflow descriptor from the provided input stream.", e);
-    } finally {
-      Closeables.closeQuietly(br);
+    }
+    if (nrOfOpenTryBlocks > 0) {
+      throw new RuntimeException("There are try-blocks missing a closing finally-block.");
```
[the remark this answers is not in the document]
Comment (mhoffrog) [reply src/main/java/com/itemis/maven/plugins/cdi/internal/util/workflow/WorkflowValidator.java:41]: The finally block opening must end with the block opening character. This is ensured by the validation.

File: src/main/java/com/itemis/maven/plugins/cdi/internal/util/workflow/WorkflowUtil.java [hunk 2]
```diff
@@ -50,85 +48,101 @@ public class WorkflowUtil {
...
-              }
+              parallelStepBuilder.addSteps(step);
             }
-          } else {
-            setDefaultExecutionData(currentStep, line);
-            setDefaultRollbackData(currentStep, line);
           }
+        } else {
+          setDefaultExecutionData(currentStep, line);
+          setDefaultRollbackData(currentStep, line);
```
[the remark this answers is not in the document]
Comment (mhoffrog) [reply src/main/java/com/itemis/maven/plugins/cdi/internal/util/workflow/WorkflowUtil.java:118]: This is ok as is. Method parseWorkflow will be left immediately when throwing RuntimeException.
